tests/pytests/unit/utils/test_cloud.py

```diff
@@ -220,3 +221,164 @@ def test_winrm_pinnned_version():
             winrm_pkg = pkg_resources.get_distribution("pywinrm")
             assert winrm_pkg.version >= '0.3.0'
     # fmt: on
+
+
+def test_ssh_gateway_arguments_default_alive_args():
+    arguments = ssh_gateway_arguments({"ssh_gateway": "host"})
+    assert "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL) in arguments
+    assert "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX) in arguments
+
+
+def test_ssh_gateway_arguments_alive_args():
+    arguments = ssh_gateway_arguments(
+        {
+            "ssh_gateway": "host",
+            "server_alive_interval": cloud.SERVER_ALIVE_INTERVAL + 1,
+            "server_alive_count_max": cloud.SERVER_ALIVE_COUNT_MAX * 2,
+        }
+    )
+    assert (
+        "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL + 1) in arguments
+    )
+    assert (
+        "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX * 2) in arguments
+    )
+
+
+@patch("salt.utils.cloud.socket", MagicMock())
+def test_wait_for_port_default_alive_args():
+    with patch(
+        "salt.utils.cloud._exec_ssh_cmd", MagicMock(return_value=0)
+    ) as exec_ssh_cmd:
```

Comment:
**suggestion** since `patch` already creates a mock, and accepts most of the same arguments, this could be simplified. Also `create_autospec/autospec=True` is super handy to ensure your tests don't get out-of-date with code:

```
with patch("salt.utils.cloud._exec_ssh_cmd", autospec=True, return_value=0) as exec_ssh_cmd:
```

Is mostly equivalent (autospec=True does change it to autospec-ing)